contracts/token/LinearVestingWithPenalty.sol

```diff
@@ -457,8 +407,28 @@ contract LinearVestingWithPenalty is Initializable, ERC20Upgradeable, AccessCont
     }
 
     /**
-     * @dev Authorizes upgrades to the contract.
-     * @param newImplementation The address of the new implementation.
-     */
+    * @dev Overrides the internal _update function to transfer vesting schedules
+    * and enforce admin-only access through the onlyAdmin modifier.
+    */
+    function _update(address from, address to, uint256 value) internal virtual override onlyRole(ADMIN_ROLE) {
```

Comment:
so, does this mean that only admin can call `.transfer()` and `.transferFrom()`?